tools/ci_build/github/apple/test_apple_packages.py

```diff
@@ -206,6 +206,12 @@ def parse_args():
         help="Prepare the test project only, without running the tests",
     )
 
+    parser.add_argument(
+        "--test_full_package_with_ios_only",
+        action="store_true",
+        help="Run iOS tests only. (Specify this argument only when package variant is Full.)",
```

Comment:
why limit it to `Full`? e.g. how about `Training`?

Reply:
currently, we only have one setting file for training variant build, https://github.com/microsoft/onnxruntime/blob/5c2837ced7bdc50dcb8d1f5c93d1df27d4689c4e/tools/ci_build/github/apple/default_training_ios_framework_build_settings.json#L1-L14

which by default contains macos. to avoid confusion/missing runs for macos platform package test, only said for full packages here.

but we can always create new configs with ios only for `Training` I suppose. Do you think it's better to loose the args for both full/training here? can update.

Reply:
making the options more general where possible gives us more flexibility later.

for example, maybe we can just have an option to skip the MacOS tests and change the logic to:

```python
if not args.skip_macos_test:
    # run MacOS test
```

alternatively, this script could parse the framework info file to determine whether the MacOS or iOS tests can be run.

Reply:
updated to use the option `skip_macos_tests`.